internal/models/linking.go

```diff
@@ -49,7 +50,19 @@ type AccountLinkingResult struct {
 // - It's not possible to decide due to data inconsistency (MultipleAccounts) and the caller should decide
 //
 // Errors signal failure in processing only, like database access errors.
-func DetermineAccountLinking(tx *storage.Connection, provider, sub string, emails []string) (AccountLinkingResult, error) {
+func DetermineAccountLinking(tx *storage.Connection, config *conf.GlobalConfiguration, emails []p.Email, aud, provider, sub string) (AccountLinkingResult, error) {
+	var verifiedEmails []string
+	var candidateEmail p.Email
+	for _, email := range emails {
+		if email.Verified || config.Mailer.Autoconfirm {
+			verifiedEmails = append(verifiedEmails, strings.ToLower(email.Email))
+		}
+		if email.Primary {
+			candidateEmail = email
+			candidateEmail.Email = strings.ToLower(email.Email)
+		}
```

Comment:
Feels like `if email.Primary` should only be considered if the email is verified?